### PR TITLE
Implement search on crossing list view

### PR DESCRIPTION
**Pull request by briaguya**
In order to implement search I:
* Added an index to the postgres schema
* Updated the crossing list view to use pagination
    * Added pagecursors to the search query
    * Replaced the list with a [react-virtualized](https://github.com/bvaughn/react-virtualized) [InfiniteLoader](https://bvaughn.github.io/react-virtualized/#/components/InfiniteLoader) and [List](https://bvaughn.github.io/react-virtualized/#/components/List) inside a [WindowScroller](https://bvaughn.github.io/react-virtualized/#/components/WindowScroller).
    * Implemented custom logic to update all cached queries correctly when updating a crossing's status

### CODE REVIEW

File: frontend/src/components/Dashboard/CrossingListPage/CrossingListItem/CrossingListItem.js [hunk 1]
```diff
@@ -39,7 +141,8 @@ class CrossingListItem extends React.Component {
       durationId: (this.state.selectedStatus === statusConstants.LONGTERM ? this.state.selectedDuration : null),
```
Comment (ifsimicoded) [inline frontend/src/components/Dashboard/CrossingListPage/CrossingListItem/CrossingListItem.js]: Not a huge issue. Can you remove the ternaries on line 140 and 141 to close out issue #76 . Same as below, I have this in my open PR, but I don't want to merge nightmare for you.

File: frontend/src/components/Dashboard/CrossingListPage/CrossingListItem/CrossingListItem.js [hunk 2]
```diff
@@ -31,6 +32,98 @@ class CrossingListItem extends React.Component {
...
+  fixSort(store, updatedCrossing) {
+    // Get the edge from the current query
+    const { crossingQueryVariables } = this.props;
+    const data = store.readQuery({ query: crossingsQuery, variables: crossingQueryVariables });
+    const edge = data.searchCrossings.edges.find(edge => edge.node.id == updatedCrossing.id);
+
+    // Get all the query variable combinations we have cached
+    const queryVariables = Object.keys(store.data.ROOT_QUERY)
+                            .filter(query => query.includes('searchCrossings'))
+                            .map(q => JSON.parse(q.replace(/.*\(|\)/g, '')));
```
Comment (benweatherman) [inline frontend/src/components/Dashboard/CrossingListPage/CrossingListItem/CrossingListItem.js]: Is this regex removing parens?

Comment (briaguya) [reply frontend/src/components/Dashboard/CrossingListPage/CrossingListItem/CrossingListItem.js]: Yes, also it's replacing the query name, so we are just parsing the variables object

screen shot 2017-12-08 at 12 50 00 pm

Comment (briaguya) [reply frontend/src/components/Dashboard/CrossingListPage/CrossingListItem/CrossingListItem.js]: screen shot 2017-12-08 at 12 59 14 pm

ooh! I see the problem, good catch!

Comment (briaguya) [reply frontend/src/components/Dashboard/CrossingListPage/CrossingListItem/CrossingListItem.js]: screen shot 2017-12-08 at 1 07 13 pm

Fixed it!

**Pull request merged**